### PR TITLE
added strings for all the combinations of categories, tags, and queries

### LINKED ISSUE #10847: Tag search queries should mention the tag name in the search results page header
Follow up for https://github.com/mozilla/addons-frontend/issues/10789#issuecomment-898459677

### Describe the problem and steps to reproduce it:
1. Try to search by query in a tag search results page - for example https://addons-dev.allizom.org/en-US/firefox/tag/facebook/?q=tab
2. Notice the search results page header

### What happened?
The search results page header doesn't mention the tag name:
image

### What did you expect to happen?
The tag name is mentioned in the search results page header

### Anything else we should know?
In a similar manner, category page search queries will mention the category name in the page header, so this is what we should be aiming for: 

https://addons-dev.allizom.org/en-US/firefox/extensions/category/bookmarks/?q=tab
image

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #10847 
I flattened some of the multiple levels of `if` statements because they become hard to follow with a 3rd level for `if (tag)`

### CODE REVIEW

**Comment by eviljeff** [comment]
Not entirely sure what I'm supposed to do with the dennis-lint CI failure or how it happened

File: src/amo/components/SearchContextCard/index.js [hunk 1]
```diff
@@ -154,7 +204,16 @@ export class SearchContextCardBase extends React.Component<InternalProps> {
           }
           break;
         default:
-          if (query) {
```
Comment (bobsilverberg) [inline src/amo/components/SearchContextCard/index.js:157]: It seems like the `category` variations are missing for this `default` case. I can see that was the case before, but is there a reason for it? I'm wondering if maybe we should fix that now.

**Review by bobsilverberg — APPROVED** [review]
Looks good, thanks @eviljeff. I just had one question about the default case.

r+wc

File: src/amo/components/SearchContextCard/index.js (hunk 1 above)
Comment (eviljeff) [reply src/amo/components/SearchContextCard/index.js:157]: categories are per add-on type, so there can't be a category for the default (unknown/missing type) case.
